### PR TITLE
BaiTap02 - TuNHT

### PR DESCRIPTION
**Pull request by at-tunguyen2**

### CODE REVIEW

File: OOP.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+
+    override func chuVi() {
+        super.chuVi()
+    }
+    
+    func dienTich() -> Double {
+        let canh1 = kichThuocCacCanh[0]
+        let canh2 = kichThuocCacCanh[1]
+        let canh3 = kichThuocCacCanh[2]
+        let p = Double((canh1 + canh2 + canh3) / 2)
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Double((canh1 + canh2 + canh3) / 2) != `Double(canh1 + canh2 + canh3)` / 2

File: OOP.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+class ClassA {
+    private var n: Int
+
+    init(_ n: Int) {
+        self.n = n
+    }
+
+    func sum() -> Int {
+        var tong = 0
+        for i in 1...n {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Please use `for i in 0..<n`

File: OOP.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+class CStack {
+    var array: [Int] = [1, 2, 3]
+
+    func isEmpty() -> Bool {
+        return array.isEmpty
+    }
+
+    func isFull() -> Bool {
+        let top = array.count
+        if top == 0 {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: What do you think if use `return top == 0`?

File: OOP.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+            return false
+        } else {
+            return true
+        }
+    }
+
+    func push(_ a: Int) {
+       array.append(a)
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Please check isFull before push in

File: OOP.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+    }
+
+    func pop() -> Int? {
+        return array.popLast()
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Please check isEmpty before pop out

File: OOP.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+        self.r = r
+        super.init(a, b)
+    }
+
+    override func hinhVe() -> String {
+        return "Hinh Tron"
+    }
+
+    override func chuVi() -> Double {
+        return (2 * Double.pi * r)
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: No need `()`

File: OOP.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+    }
+
+    override func dienTich() -> Double {
+        return (Double.pi * pow(r, 2))
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: No need `()`

File: OOP.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+    override init(_ a: Double, _ b: Double) {
+        super.init(a, b)
+    }
+
+    override func hinhVe() -> String {
+        return "Hinh Vuong"
+    }
+
+    override func chuVi() -> Double {
+        return (a * 4)
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: No need `()`

File: OOP.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+        return (2 * Double.pi * r)
+    }
+
+    override func dienTich() -> Double {
+        return (Double.pi * pow(r, 2))
+    }
+}
+
+class HinhVuong: HaiChieu {
+    override init(_ a: Double, _ b: Double) {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: If you use only, no need to override

File: OOP.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,534 @@
...
+        self.b = b
+    }
+
+    func hinhVe() -> String {
+        return "Chua xac dinh duoc hinh ve"
+    }
+}
+
+class HaiChieu: HinhVe {
+    override init(_ a: Double, _ b: Double) {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: If you use only, no need to override

File: OOP.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,534 @@
...
+        self.c = c
+        super.init(a, b)
+    }
+
+    override func hinhVe() -> String {
+        return "Hinh Tam Giac"
+    }
+
+    override func chuVi() -> Double {
+        return (a + b + c)
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: No need `()`

File: OOP.playground/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,534 @@
...
+    override init(_ a: Double, _ b: Double, _ c: Double) {
+        super.init(a, b, c)
+    }
+
+    override func hinhVe() -> String {
+        return "Hinh Cau"
+    }
+
+    override func dienTich() -> Double {
+        return (4 * Double.pi * pow(a, 2))
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: No need `()`

File: OOP.playground/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,534 @@
...
+    override init(_ a: Double, _ b: Double, _ c: Double) {
+        super.init(a, b, c)
+    }
+
+    override func hinhVe() -> String {
+        return "Hinh Lap Phuong"
+    }
+
+    override func dienTich() -> Double {
+        return (6 * pow(a, 2))
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: No need `()`

File: OOP.playground/Contents.swift [hunk 14]
```diff
@@ -0,0 +1,534 @@
...
+        for i in array {
+            print(i)
+        }
+        return 0
+    }
+
+    private var max = 0
+    private var min = 9
+
+    func printMaxMin () -> (Int, Int) {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Remove space

File: OOP.playground/Contents.swift [hunk 15]
```diff
@@ -0,0 +1,534 @@
...
+class DanhSachHocSinh {
+    private var danhSach: [HocSinh]
+
+    init(_ danhSach: [HocSinh]) {
+        self.danhSach = danhSach
+    }
+
+    func sapXep() -> [HocSinh] {
+        for i in 0..<danhSach.count {
+            for j in 1..<danhSach.count - i {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Please use `for j in 0..`

File: OOP.playground/Contents.swift [hunk 16]
```diff
@@ -0,0 +1,534 @@
...
+                if danhSach[j].tongDiem < danhSach[j-1].tongDiem {
+                    let x = danhSach[j]
+                    danhSach[j] = danhSach[j-1]
+                    danhSach[j-1] = x
+                } else if danhSach[j].tongDiem == danhSach[j-1].tongDiem {
+                    for i in 1..<danhSach.count {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Please use `for i in 0..`, `for j in 0..`

File: OOP.playground/Contents.swift [hunk 17]
```diff
@@ -0,0 +1,534 @@
...
+        if month > 12 || month < 1 {
+            return nil
+        } else {
+            switch month {
+            case 1, 3, 5, 7, 8, 10, 12:
+                return 31
+            case 4, 6, 9, 11:
+                return 30
+            case 2:
+                if year % 4 == 0 {
```
Comment (at-bienle) [inline OOP.playground/Contents.swift]: Wrong condition

File: OOP.playground/Contents.swift (hunk 15 above)
Comment (at-tunguyen2) [reply OOP.playground/Contents.swift]: I alreddy changed this but it will cause an error

File: OOP.playground/Contents.swift (hunk 16 above)
Comment (at-tunguyen2) [reply OOP.playground/Contents.swift]: I alreddy changed this but it will cause an error

**Pull request merged**